wcomponents-theme/src/main/js/wc/ui/tabset.js

```diff
@@ -464,32 +464,33 @@ function(toArray, ariaAnalog, formUpdateManager, getFilteredGroup, initialise, s
 		 * @function module:wc/ui/tabset.shedObserver
 		 * @protected
 		 * @override
-		 * @param {Element} element The element on which the shed action acted.
-		 * @param {String} action The type of shed event. One of EXPAND, COLLAPSE, SELECT or DESELECT.
+		 * @param {Event} $event The shed event that fired.
 		 */
-		this.shedObserver = function (element, action) {
+		this.shedObserver = function ($event) {
+			var element = $event.target,
+				action = $event.type;
 			if (element) {
 				if (this.ITEM.isOneOfMe(element)) {
 					switch (action) {
-						case shed.actions.SELECT:
-						case shed.actions.DESELECT:
+						case shed.events.SELECT:
+						case shed.events.DESELECT:
 							onItemSelection(action, element);
 							break;
-						case shed.actions.EXPAND:
-						case shed.actions.COLLAPSE:
+						case shed.events.EXPAND:
+						case shed.events.COLLAPSE:
 							onItemExpansion(action, element);
 							break;
-						case shed.actions.DISABLE:
+						case shed.events.DISABLE:
 							onItemDisabled(element);
 							break;
-						case shed.actions.ENABLE:
+						case shed.events.ENABLE:
 							onItemEnabled(element);
 							break;
 						default:
 							console.warn("Unknown action", action);
 							break;
 					}
-				} else if ((action === shed.actions.DISABLE || action === shed.actions.ENABLE) && TABLIST.isOneOfMe(element)) {
+				} else if ((action === shed.events.DISABLE || action === shed.events.ENABLE) && TABLIST.isOneOfMe(element)) {
 					// if the tablist is disabled or enabled, diable/enable all the tabs.
 					Array.prototype.forEach.call(this.ITEM.findDescendants(element), function (next) {
 						shed[action](next);
```

Comment:
Would the above line ```shed[action](next)``` be OK. Although no tests are failing, I think that would end up being something like ```shed['wcdisable'](next)```. There's no ```wcdisable``` method on shed, so should it be mapped back to ```disable``` or ```enable```.

Reply:
I think ```shed.events[action]``` would map it back.

Reply:
Yes this looks like it won't work as is - mapping it back would be an easy solution